### DISCUSSION TITLE
How to Improve AIdeck WiFi Speed

### DISCUSSION THREAD

**Discussion by mfxjx000205** [bitcraze/discussions#908] [Q&A]
Hi, everyone.

I'm trying to improve the AIdeck's WiFi speed in order to fix my experiment. I have built a program in GAP8, which can always send 64KB packet to WIFI_HOST, like wifi-img-streamer. And I have also built a program in python, to receive these 64KB packet via socket. I measure the WiFi speed, which can reach **1MBps** over air and almost **600KBps** through socket when the distance is 2 meters. When I run the wifi-img-streamer, the result is same. 

I want to ask that: 

1. Can we improve the WiFi speed via esp-idf's menuconfig or aideck-esp-firmware? for example: buffer size, cpu frequence, dbm, channels, and so on.

2. Can we use UDP for data transmission. Is there anything I need to do in firmware?

By the way, I read the NINA-W102 datasheet, It seems that the WiFi speed can reach at least 6.5MBps in 802.11n model. There is a huge gap between 1MBps and 6.5MBps. 

Bset wishes,

Hanjie

**Comment by evoggy** [comment]
Hi!

Yes, there's probably more to be done in order to improve the performance of the WiFi.

1) I think there's tweaks to be done here, but unfortunately I can't say which ones. There's a lot of parameters for the stack which can be tuned. I would recommend tuning a few and seeing if anything changes. It's also an option to update to a later version of the ESP IDF to see if this makes any changes.
2) Yes, I think both the Python code and the ESP32 code needs to be updated for this

I looked into the intermittent performance issues I was seeing a while ago, but unfortunately I didn't get very far. From what I could see the sending of data suddenly takes a lot longer and I do not know why. The call to the library to send the data would suddenly take a lot longer to return and then all the buffers for CPX filled up all the way back to the GAP8.
